File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/string/StringIsContainedRule.java
package com.example.navigation.stepsEngine.flow.rules.comparable.string;

import com.example.navigation.stepsEngine.flow.rules.comparable.ComparableRule;
import com.example.navigation.stepsEngine.payment.FlowState;

public class StringIsContainedRule extends ComparableRule {

    private String value;
    private Integer start;
    private Integer end;

    public Integer getStart() {
        return start;
    }

    public void setStart(Integer start) {
        this.start = start;
    }

    public Integer getEnd() {
        return end;
    }

    public void setEnd(Integer end) {
        this.end = end;
    }

    @Override
    public boolean evaluate(FlowState flowState) {
        int from = start == null ? 0 : start;
        final String string = (String) getValue(flowState);
        int to = end == null ? string.length() : end;
        final String s = string.substring(from, to);
        return value.contains(s);
    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/string/StringIsContainedRule.kt
package com.example.navigation.stepsEngine.flow.rules.comparable.string

import com.example.navigation.stepsEngine.flow.rules.comparable.ComparableRule
import com.example.navigation.stepsEngine.payment.FlowState

class StringIsContainedRule : ComparableRule() {
    var start: Int? = null
    var end: Int? = null

    override fun evaluate(flowState: FlowState): Boolean {
        val from = if (start == null) 0 else start!!
        val string = getValue(flowState) as String?
        val to = if (end == null) string!!.length else end!!
        val s = string!!.substring(from, to)
        return value!!.contains(s)
    }
}